### PR TITLE
Devour Tweak 2.0

### PR DESCRIPTION
**Pull request by Surrealaser**
-Devoured marines are now blinded for the duration of the devour, allowing aliens the opportunity to deposit them in unrecognizable locations.

-Devour delay now scales inversely with the pain the victim is in; the higher the pain (i.e. they're less able to resist), the faster it is completed, with a current max delay of 20 seconds vs a full health/no pain marine, and a minimum delay of 3 seconds. The delay is reduced by 1 second per 10 pain.

### CODE REVIEW

**Comment by subject217** [comment]
fetish content

**Comment by Rohesie** [comment]
Conflicts as expected.

File: code/modules/mob/mob_grab.dm [hunk 1]
```diff
@@ -97,7 +97,7 @@
 						return */
 		X.visible_message("<span class='danger'>[X] starts to devour [pulled]!</span>", \
 		"<span class='danger'>You start to devour [pulled]!</span>", null, 5)
-		if(do_after(X, 50, FALSE, 5, BUSY_ICON_HOSTILE))
+		if(do_after(X, min(30, 200 - pulled.traumatic_shock), FALSE, 5, BUSY_ICON_HOSTILE)) //Devour delay scales inversely with victim's pain; 20 seconds max, 3 seconds min
```
Comment (Rohesie) [inline code/modules/mob/mob_grab.dm]: I think you want here a `max(30, 200- pulled.traumatic_shock)`, no? `CLAMP` would be safer.

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon_helpers.dm [hunk 2]
```diff
@@ -10,3 +10,31 @@
...
+		A.forceMove(loc)
+		if(ismob(A))
+			if(forced)
+				visible_message("<span class='danger'>[A] bursts out of [src]!</span>")
+			if(isliving(A))
+				var/mob/living/L = A
+				L.devoured = FALSE //remove the devour flag
+	return TRUE
+
+/mob/living/carbon/proc/unvore_forced(var/mob/living/carbon/Xenomorph/X)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_helpers.dm]: This seems to overlap with `unvore()` with a `forced` flag.

File: code/modules/mob/living/carbon/xenomorph/XenoProcs.dm [hunk 3]
```diff
@@ -524,3 +513,17 @@
 		leader_aura_strength = Q.aura_strength
 		leader_current_aura = Q.current_aura
 		to_chat(src, "<span class='xenowarning'>Your pheromones have changed. The Queen has new plans for the Hive.</span>")
+
+//Bleuugh
+/mob/living/carbon/Xenomorph/proc/empty_gut()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/XenoProcs.dm]: This overlaps as well.

**Review by Rohesie — CHANGES_REQUESTED** [review]

**Pull request closed without merging**